cylc/flow/cycling/__init__.py

```diff
@@ -365,10 +365,9 @@ def set_offset(self, i_offset):
         """Deprecated: alter state to offset the entire sequence."""
         pass
 
-    @abstractmethod
     def is_on_sequence(self, point):
         """Is point on-sequence, disregarding bounds?"""
-        pass
+        raise NotImplementedError
```

Comment:
Why? I thought `@asbtractmethod` is the correct thing to do?

Reply:
"Correct" is subjective, both are valid approaches, asbtractmethod is more OO, NotImplementedError is more functional. Python supports both paradigms but they can clash.

OO approaches are often less flexible, in this case the `asbtractmethod` mandated that this method must exist in the child class. The solution on this PR uses the flexibility of Python's functional side to add the method at `__init__` time, OO just doesn't like this.

Reply:
I think it's probably adding a comment about why we can't use `@abstractmethod`

Reply:
Done - https://github.com/cylc/cylc-flow/pull/4747/files#r834249914